src/tiled/commandmanager.cpp

```diff
@@ -59,17 +59,32 @@ CommandDataModel *CommandManager::commandDataModel()
     return mModel;
 }
 
+void CommandManager::registerMenu(QMenu *menu)
+{
+    mMenus.append(menu);
+    menu->clear();
+    menu->addActions(mActions);
+}
+
 void CommandManager::showDialog()
 {
     CommandDialog dialog(QApplication::activeWindow());
     dialog.exec();
 }
 
-void CommandManager::populateMenu(QMenu *menu, bool flag)
+void CommandManager::populateMenus()
 {
-    menu->clear();
+    for (int i = 0; i < mMenus.size(); ++i) {
+        QMenu *menu = mMenus.at(i);
```

Comment:
@bjorn, I'm a bit confused here. In `updateActions` too, we didn't actually use the indices. Why didn't you suggest using this structure there as well?